### PR TITLE
added movie list filter

### PR DESCRIPTION
**Pull request by silchyn**
[DEMO](https://silchyn.github.io/react_movies-list-filter/)

### CODE REVIEW

**Review by LordAsmodey — APPROVED** [review]
Looks good

File: src/App.tsx [hunk 1]
```diff
@@ -1,9 +1,19 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const visibleMovies = useMemo(() => {
```
Comment (Ihor-Karpyn) [inline src/App.tsx:8]: add an empty line

File: src/App.tsx [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
+    return moviesFromServer.filter(movie => {
+      return movie.title.toLowerCase()
+        .includes(query.toLowerCase())
+        || movie.description.toLowerCase()
+          .includes(query.toLowerCase());
+    });
```
Comment (Ihor-Karpyn) [inline src/App.tsx:14]: Move filter to separate function. Hard for understanding

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
+  }, [query]);
```
Comment (Ihor-Karpyn) [inline src/App.tsx:15]:
```suggestion
  const visibleMovies = useMemo(() => moviesFromServer.filter(movie => (
     movie.title.toLowerCase().includes(query.toLowerCase())
     || movie.description.toLowerCase().includes(query.toLowerCase())
  )), [query]);
```

**Review by Ihor-Karpyn — COMMENTED** [review]
Or smth like that

File: src/App.tsx [hunk 4]
```diff
@@ -1,9 +1,20 @@
...
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
+const filterMovies = (query: string) => {
+  return () => moviesFromServer.filter(movie => movie.title.toLowerCase()
+    .includes(query.toLowerCase())
+      || movie.description.toLowerCase()
+        .includes(query.toLowerCase()));
+};
```
Comment (Ihor-Karpyn) [inline src/App.tsx:11]:
```suggestion
const filterMovies = (query: string) => (moviesFromServer.filter(movie => {
	const preparedQuery = query.toLowerCase();
	const title = movie.title.toLowerCase();
	const description = movie.description.toLowerCase();

  const isTitleMatched = title.includes(preparedQuery))
  const isDescriptionMatched = description.includes(preparedQuery));
  
  return isTitleMatched || isDescriptionMatched
});
```

File: src/App.tsx [hunk 5]
```diff
@@ -1,9 +1,20 @@
...
+
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const visibleMovies = useMemo(filterMovies(query), [query]);
```
Comment (Ihor-Karpyn) [inline src/App.tsx:16]:
```suggestion
  const visibleMovies = useMemo(() => filterMovies(query), [query]);
```

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]
Good job, but filter function still very ard

**Review by Ihor-Karpyn — APPROVED** [review]
I like it, much better 🚀

**Pull request closed without merging**